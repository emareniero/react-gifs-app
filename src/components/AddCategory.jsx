import { useState } from "react";
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {
  
  const [inputValue, setInputValue] = useState('');

  const onInputChange = (ev) => {
    // desestructuramos value del evento!
    const { value } = ev.target;
    // Lo agregamos para que react se entere y cambie el estado
    setInputValue(value);
  };

  const onEventSubmit = (ev) => {
    // Evitamos que se vuelva a cargar toda la pagina
    ev.preventDefault();
    
    // Chequeamos que no se envien espacios vacios
    if (inputValue.trim().length <= 1) return;

    // Limpiamos form
    setInputValue('');
    
    // Como ahora recibimos la funcion setCateogries
    // le agregamos el valor que estamos tipeadndo
    onNewCategory( inputValue.trim() );
    
  };

  return (
    <form onSubmit={onEventSubmit} aria-label="form" >
      <input 
        type="text" 
        placeholder="Buscar Gifs!" 
        value={ inputValue }
        onChange={ onInputChange } />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}
