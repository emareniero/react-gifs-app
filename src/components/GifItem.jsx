// Importamos el prop-types
import PropTypes from 'prop-types';


export const GifItem = ({ title, img }) => {

  return (
    <div className="card">

        <img src={img} alt={title} />
        <p> {title} </p>

    </div>
  )
}

// Ponemos las caracter√≠sticas que tienen que tener las prop types.
GifItem.propTypes = {

  title: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired

}