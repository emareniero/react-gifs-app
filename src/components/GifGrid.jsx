import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {
  // desestructuramos las images y isLoading de nuestri fetch
  const { images, isLoading } = useFetchGifs(category);

  // devolvemos la respuesta del componente
  return (
    <>
      <h3> {category} </h3>
      {
        isLoading && <h2>Cargando...</h2>
      }

      <div className="card-grid">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}
