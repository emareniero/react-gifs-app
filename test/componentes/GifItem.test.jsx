import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Test al componente <GifItem/>", () => {
  // Creamos algunas instancias para mandarle a nuestro componente
  const title = "Goku";
  const url = "https://www.goku.com/images";

  test("debe hacer match con el Snapshot", () => {
    // Tomamos el container del componente para tomarle una imagen
    const { container } = render(<GifItem title={title} img={url} />);

    // Sacamos una foto para que si luego el componente cambia nos avise
    expect(container).toMatchSnapshot();
  });

  test("debe mostrar la imágen con el URL(img) y el ALT indicado", () => {
    // Renderizamos el componente que queremos testear
    render(<GifItem title={title} img={url} />);

    // Si queremos ver que estamos renderizando hacemos uso de screen.debug()
    // screen.debug();

    // Chequeamos que la imagen que esta mandando sea exacamente la que nosotros queremos
    // expect( screen.getByRole( 'img' ).src ).toBe( url )
    // console.log(screen.getByRole( 'img' ))

    // Para hacer mas simple el codigo podemos desestructrar lo que queremos testear
    const { src, alt } = screen.getByRole("img");

    // Testeamos que venga título e imágen
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("debe mostrar el título en el componente", () => {
    // Renderizamos el componente que queremos testear
    render(<GifItem title={title} img={url} />);

    // Chequeamos que venga el titulo
    expect(screen.getByText(title)).toBeTruthy();
  });
});
