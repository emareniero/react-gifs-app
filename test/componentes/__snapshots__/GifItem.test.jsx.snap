// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test al componente <GifItem/> debe hacer match con el Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://www.goku.com/images"
    />
    <p>
       
      Goku
       
    </p>
  </div>
</div>
`;
